crates/astria-conductor/src/data_availability.rs

```diff
@@ -333,37 +349,42 @@ async fn verify_sequencer_blobs_and_assemble_rollups(
 /// Fetches rollup blobs for the given height and namespace, verifiying it
 /// against the sequencer "header" blob.
 async fn fetch_and_verify_rollup_blobs(
-    client: CelestiaClient,
+    client: HttpClient,
     height: u64,
     data: SignedNamespaceData<SequencerNamespaceData>,
     namespace: Namespace,
     block_tx: mpsc::Sender<SequencerBlockSubset>,
 ) {
-    let rollup_data = match client.get_rollup_data(height, &data, namespace).await {
+    let rollups = match client
+        .get_rollup_data_matching_sequencer_data(height, namespace, &data)
+        .await
+    {
         Err(e) => {
             warn!(error.message = %e, error.cause = ?e, "failed to get rollup data from celestia");
             return;
         }
-        Ok(None) => {
+        Ok(rollups) if rollups.is_empty() => {
             info!("could not find rollup data");
             return;
         }
-        Ok(Some(rollup_data)) => rollup_data,
+        Ok(rollups) => rollups,
     };
 
-    if block_verifier::validate_rollup_data(&rollup_data, data.data.action_tree_root).is_err() {
-        warn!(
-            "could not validate rollup data given the action tree root of the sequencer data; \
-             dropping the block"
-        )
-    } else {
-        let subset = SequencerBlockSubset {
-            block_hash: data.data.block_hash,
-            header: data.data.header,
-            rollup_transactions: rollup_data.rollup_txs,
-        };
-        if block_tx.send(subset).await.is_err() {
-            warn!("failed sending validated rollup data to block assembler; receiver dropped");
+    for rollup in rollups {
+        if block_verifier::validate_rollup_data(&rollup, data.data().action_tree_root).is_err() {
+            warn!(
+                "could not validate rollup data given the action tree root of the sequencer data; \
+                 dropping the block"
+            )
```

Comment:
i guess this case would catch it - there is only one rollup data that can pass this check

Reply:
Yeah, so just in case this doesn't happen it will now noisily drop those cases where more than 1 blob remains after verification.